frontend/src/core/api/base.js

```diff
@@ -87,4 +87,39 @@ export default class APIBase {
             return {};
         }
     }
+
+    keysToCamelCase(results) {
+        const toCamelCase = (s) => {
+            return s.replace(/([-_][a-z])/ig, ($1) => {
+              return $1.toUpperCase()
+                .replace('-', '')
+                .replace('_', '');
+            });
+        };
+
+        const isArray = function (arr) {
+            return Array.isArray(arr);
+        };
+
+        const isObject = function (obj) {
+            return obj === Object(obj) && !isArray(obj) && typeof obj !== 'function';
+        };
```

Comment:
Can you move `toCamelCase` and `isObject` to outside this method? Maybe at the top of this file? 

I'm not exactly sure how compilers optimize this, but there's a risk that these functions will be re-created every time `keysToCamelCase` is called, taking unnecessary memory and execution time. It's good practice to declare functions inside functions only when strictly needed. :-)